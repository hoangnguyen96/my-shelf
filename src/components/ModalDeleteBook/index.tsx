import {
  Modal,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
} from "@chakra-ui/react";
import { Button } from "../common";
import { memo } from "react";

interface ModalDeleteBookProps {
  isOpen?: boolean;
  onClose: () => void;
  onConfirm: () => void;
}

const ModalDeleteBook = ({
  isOpen = false,
  onClose,
  onConfirm,
}: ModalDeleteBookProps) => (
  <Modal isOpen={isOpen} onClose={onClose} isCentered>
    <ModalOverlay />
    <ModalContent p="40px" minW={484} alignItems="center">
      <ModalHeader textAlign="center">
        Are you sure to delete this book?
      </ModalHeader>

      <ModalFooter gap="20px">
        <Button
          data-testid="confirm-delete"
          onClick={onConfirm}
          text="Yes"
          size="md"
          variant="outline"
        />
        <Button onClick={onClose} text="No" size="md" variant="outline" />
      </ModalFooter>
    </ModalContent>
  </Modal>
);

export default memo(ModalDeleteBook);
