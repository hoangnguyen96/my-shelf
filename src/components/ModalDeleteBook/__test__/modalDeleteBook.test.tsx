import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import ModalDeleteBook from "..";

describe("ModalDeleteBook", () => {
  it("Should render correctly snapshot", () => {
    expect(
      render(
        <ModalDeleteBook
          isOpen={true}
          onClose={() => {}}
          onConfirm={() => {}}
        />
      )
    ).toMatchSnapshot();
  });
});
