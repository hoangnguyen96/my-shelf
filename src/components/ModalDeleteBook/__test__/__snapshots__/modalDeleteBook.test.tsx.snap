// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDeleteBook Should render correctly snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    data-scroll-locked="1"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    />
    <div
      class="chakra-portal"
    >
      <div
        aria-hidden="true"
        class="chakra-modal__overlay css-13d1tkw"
        data-aria-hidden="true"
        style="opacity: 0; will-change: opacity;"
      />
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
      <div
        data-focus-lock-disabled="false"
      >
        <div
          class="chakra-modal__content-container css-1oxhx2p"
          tabindex="-1"
        >
          <section
            aria-labelledby="chakra-modal--header-:r0:"
            aria-modal="true"
            class="chakra-modal__content css-gzck6b"
            id="chakra-modal-:r0:"
            role="dialog"
            style="opacity: 0; will-change: opacity,transform; transform: scale(0.95);"
            tabindex="-1"
          >
            <header
              class="chakra-modal__header css-3scr2j"
              id="chakra-modal--header-:r0:"
            >
              Are you sure to delete this book?
            </header>
            <footer
              class="chakra-modal__footer css-1pv6r1e"
            >
              <button
                class="chakra-button css-4xx2wk"
                type="button"
              >
                Yes
              </button>
              <button
                class="chakra-button css-4xx2wk"
                type="button"
              >
                No
              </button>
            </footer>
          </section>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
